src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -111,52 +111,65 @@ export default class AircraftController {
      * This is the entry method for creating new departing and arriving aircraft.
      * This method should only be called as a callback from a `SpawnScheduler` timer.
      *
-     * @method createAircraftWithSpawnModel
+     * @method createAircraftWithSpawnPatternModel
      * @param spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    createAircraftWithSpawnModel = (spawnPatternModel) => {
+    createAircraftWithSpawnPatternModel = (spawnPatternModel) => {
         const initializationProps = this._buildAircraftProps(spawnPatternModel);
-        const aircraftModel = new AircraftInstanceModel(initializationProps);
 
-        this.addItem(aircraftModel);
+        this._createAircraftWithInitializationProps(initializationProps);
     }
 
     /**
-     * @for AircraftController
-     * @method aircraft_auto_toggle
+     * Build aircraft props for `spawnPatternModel` and add
+     * preSpawn data to `baseAircraftProps`.
+     *
+     * Used when creating aircraft already along an arrival route on load
+     * or on airport change.
+     *
+     * This method should be called direectly and not via a timer callback
+     *
+     * This hooks into the same method used to build spawnning aircraft
+     * and simply adds another layer on top of that to build a preSpawn aircraft.
+     *
+     * @method createPreSpawnAircraftWithSpawnPatternModel
+     * @param  spawnPatternModel {SpawnPatternModel}
+     * @private
      */
-    aircraft_auto_toggle() {
-        this.aircraft.auto.enabled = !this.aircraft.auto.enabled;
-    }
+    createPreSpawnAircraftWithSpawnPatternModel = (spawnPatternModel) => {
+        for (let i = 0; i < spawnPatternModel.preSpawnAircraftList.length; i++) {
```

Comment:
normally I'd go with a `_forEach` here (of which `_each` is an alias of).  However, during load speed is important.  I don't have anything to base this off of, but I'm convinced a native for loop is going to be the fastest way through this data.

At the end of the day, though, I could just as easily use `_forEach` but I'd bet native `for` is faster.